test/geometry/lane_geometry_test.cc

```diff
@@ -0,0 +1,193 @@
+  for (std::size_t i = 0; i < case_.p.size(); ++i) {
+    const auto rpy = lane_geometry.Orientation(case_.p[i]);
+    EXPECT_EQ(case_.expected_rpy[i].vector(), rpy.vector())
+        << "Expected RPY: " << case_.expected_rpy[i].vector() << " vs RPY: " << rpy.vector();
+  }
+}
+
+INSTANTIATE_TEST_CASE_P(OrientationTestGroup, OrientationTest, ::testing::ValuesIn(OrientationTestCases()));
+
+struct WTestCase {
+  LineString3d left{};
+  LineString3d right{};
+  std::vector<Vector3> prh{};
+  std::vector<Vector3> expected_w{};
+};
+
+std::vector<WTestCase> WTestCases() {
+  return {
+      {
+          // No elevation along x
+          LineString3d{{0., 2., 0.}, {100., 2., 0.}} /* left*/,
+          LineString3d{{0., -2., 0.}, {100., -2., 0.}} /* right*/,
+          {{0., 0., 0.}, {50., 0., 0.}, {100., 0., 0.}, {0., 4., 2.}, {50., -2., -2.}, {100., 7., -1.}} /* prh */,
+          {{0., 0., 0.}, {50., 0., 0.}, {100., 0., 0.}, {0., 4., 2.}, {50., -2., -2.}, {100., 7., -1.}} /* expected_w */
+      },
+      {
+          // No elevation along -x
+          LineString3d{{0., 2., 0.}, {-100., 2., 0.}} /* line string*/,
+          LineString3d{{0., -2., 0.}, {-100., -2., 0.}} /* line string*/,
+          {{0., 0., 0.}, {50., 0., 0.}, {100., 0., 0.}, {0., 5., 1.}, {50., -6., -12.}, {100., -24., 4.}} /* prh */,
+          {{0., 0., 0.},
+           {-50., 0., 0.},
+           {-100., 0., 0.},
+           {0., -5., 1.},
+           {-50., 6., -12.},
+           {-100., 24., 4.}} /* expected_w
+                              */
+      },
+      {
+          // Linear Elevation. `h` value implies a movement in x.
+          LineString3d{{0., 2., 0.}, {100., 2., 100.}} /* left*/,
+          LineString3d{{0., -2., 0.}, {100., -2., 100.}} /* right*/,
+          {{0., 0., 0.},
+           {std::sqrt(2.) * 50., 0., 0.},
+           {std::sqrt(2.) * 100., 0., 0.},
+           {0., 5., 0.},
+           {std::sqrt(2.) * 50., 0., 5.},
+           {std::sqrt(2.) * 100., 5., 5.}} /* prh */,
+          {{0., 0., 0.},
+           {50., 0., 50.},
+           {100., 0., 100.},
+           {0., 5., 0.},
+           {50. - 5 * std::sqrt(2.) / 2., 0., 50. + 5 * std::sqrt(2.) / 2.},
+           {100. - 5 * std::sqrt(2.) / 2., 5., 100. + 5 * std::sqrt(2.) / 2.}} /* expected_w */
+      },
+      {
+          // Linear Elevation with negative yaw. `h` value implies a movement in x and y.
+          LineString3d{{1., 1., 0.}, {1. + 70.71067811865476, 1. - 70.71067811865476, 100.}} /* left*/,
+          LineString3d{{-1., -1., 0.}, {-1. + 70.71067811865476, -1. - 70.71067811865476, 100.}} /* right*/,
+          {{0., 0., 0.}, {141.4213562373095, 0., 0.}, {0., 5., -2.}} /* prh */,
+          {{0., 0., 0.},
+           {70.71067811865476, -70.71067811865476, 100.},
+           {0. + 5 * std::sqrt(2.) / 2. + 2 * std::sqrt(2.) / 2., 0. + 5 * std::sqrt(2.) / 2. - 2 * std::sqrt(2.) / 2.,
+            -2. * std::sqrt(2.) / 2.}} /* expected_w
+                                        */
```

Comment:
It is failing for this case, trying to find the bug 

```
/home/franco/maliput_ws_focal/src/maliput_sparse/test/geometry/lane_geometry_test.cc:185: Failure
Value of: maliput::math::test::CompareVectors(case_.expected_w[i], dut, kTolerance)
  Actual: false (Value at (0) exceeds tolerance: 4.9497474683058327 vs. 4.5355339059327378, diff = 0.41421356237309492, tolerance = 9.9999999999999998e-13
v1 =
{4.94974746830583, 2.12132034355964, -1.4142135623731}
v2 =
{4.53553390593274, 2.53553390593274, -1.4142135623731}
delta=
{0.414213562373095, -0.414213562373095, 0})
Expected: true

```

Reply:
I will take another look at it later today.